src/clean.c

```diff
@@ -288,15 +309,28 @@ static int clean_staged_manifests(const char *path, bool dry_run, bool all)
 			continue;
 		}
 
+		char *version_dir;
+		string_or_die(&version_dir, "%s/%s", state_dir, name);
+
+		/* Remove all hash-hint manifest files.
+		 * Don't perform mom_contents check because we always want to remove
+		 * these */
+		ret = remove_if(version_dir, dry_run, is_hashed_manifest);
```

Comment:
When --all, --dry-run is going to print hashed manifests twice and we will check for the prefix "Manifest." twice for all files. I prefer to run remove_if only once, using is_manifest when name isn't in mom_contents and is_hashed_manifest otherwise.

Reply:
ah, good catch.